### PR TITLE
feat: Allow overriding service-account-name

### PR DESCRIPTION
**Pull request by nitrocode**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

- Resolves #<issue-number>

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [x] E2E Test successfully complete before merge?

### Additional Notes

Previous PR https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1184

### CODE REVIEW

File: modules/kubernetes-addons/appmesh-controller/main.tf [hunk 1]
```diff
@@ -38,7 +39,7 @@ module "helm_addon" {
     create_kubernetes_namespace       = true
     kubernetes_namespace              = local.namespace
     create_kubernetes_service_account = true
-    kubernetes_service_account        = local.name
+    kubernetes_service_account        = try(var.helm_config.service_account, local.name)
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/appmesh-controller/main.tf:42]:
```suggestion
    kubernetes_service_account        = local.service_account_name
```

File: modules/kubernetes-addons/appmesh-controller/main.tf [hunk 2]
```diff
@@ -1,6 +1,7 @@
 locals {
-  name      = try(var.helm_config.name, "appmesh-controller")
-  namespace = try(var.helm_config.namespace, "appmesh-system")
+  name                 = try(var.helm_config.name, "appmesh-controller")
+  namespace            = try(var.helm_config.namespace, "appmesh-system")
+  service_account_name = try(var.helm_config.service_account, local.name)
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/appmesh-controller/main.tf:4]: lets standardize as much as we can on `service_account_name`, where it doesn't break the API compatibility

File: modules/kubernetes-addons/adot-collector-haproxy/main.tf [hunk 3]
```diff
@@ -1,6 +1,7 @@
 locals {
-  name      = "adot-collector-haproxy"
-  namespace = try(var.helm_config.namespace, local.name)
+  name            = "adot-collector-haproxy"
+  namespace       = try(var.helm_config.namespace, local.name)
+  service_account = try(var.helm_config.service_account_name, local.name)
```
Comment (nitrocode) [inline modules/kubernetes-addons/adot-collector-haproxy/main.tf:4]: This is a breaking change so I will revert.

File: modules/kubernetes-addons/adot-collector-java/main.tf [hunk 4]
```diff
@@ -1,6 +1,7 @@
 locals {
-  name      = "adot-collector-java"
-  namespace = try(var.helm_config.namespace, local.name)
+  name                 = "adot-collector-java"
+  namespace            = try(var.helm_config.namespace, local.name)
+  service_account_name = try(var.helm_config.service_account_name, local.name)
```
Comment (nitrocode) [inline modules/kubernetes-addons/adot-collector-java/main.tf:4]: This is a breaking change so I will revert.

File: modules/kubernetes-addons/adot-collector-memcached/main.tf [hunk 5]
```diff
@@ -1,6 +1,7 @@
 locals {
-  name      = "adot-collector-memcached"
-  namespace = try(var.helm_config.namespace, local.name)
+  name                 = "adot-collector-memcached"
+  namespace            = try(var.helm_config.namespace, local.name)
+  service_account_name = try(var.helm_config.service_account_name, local.name)
```
Comment (nitrocode) [inline modules/kubernetes-addons/adot-collector-memcached/main.tf:4]: This is a breaking change so I will revert.

File: modules/kubernetes-addons/adot-collector-nginx/main.tf [hunk 6]
```diff
@@ -1,6 +1,8 @@
 locals {
   name      = "adot-collector-nginx"
   namespace = try(var.helm_config.namespace, local.name)
+
+  service_account_name = try(var.helm_config.service_account_name, local.name)
```
Comment (nitrocode) [inline modules/kubernetes-addons/adot-collector-nginx/main.tf:5]: This is a breaking change so I will revert.

File: modules/kubernetes-addons/aws-load-balancer-controller/README.md [hunk 7]
```diff
@@ -62,7 +62,7 @@ If the IAM role is too long, override the service account name in the `helm_conf
 ```hcl
   enable_aws_load_balancer_controller = true
   aws_load_balancer_controller_helm_config = {
-    service_account_name = "aws-lb-sa"
+    service_account = "aws-lb-sa"
```
Comment (nitrocode) [inline modules/kubernetes-addons/aws-load-balancer-controller/README.md:65]:
```suggestion
    service_account_name = "aws-lb-sa"
```

File: modules/kubernetes-addons/appmesh-controller/main.tf (hunk 2 above)
Comment (nitrocode) [reply modules/kubernetes-addons/appmesh-controller/main.tf]: @bryantbiggs I addressed this change by allowing both `service_account` and `service_account_name` for the `adot*` modules.

All the others only allow `service_account_name`.

Comment (nitrocode) [reply modules/kubernetes-addons/appmesh-controller/main.tf]: @bryantbiggs please review when you get a chance. I'm hoping to have this merged prior to the next release (pretty please) :D

Comment (bryantbiggs) [reply modules/kubernetes-addons/appmesh-controller/main.tf]: my apologies - I didn't take the proper time to look and mislead you to rename incorrectly. It looks like most of the names used are `service_account` or `kubernetes_service_account` so we should use `service_account` and not `service_account_name`. I think I've caught all the instances, will merge once CI passes green

Comment (nitrocode) [reply modules/kubernetes-addons/appmesh-controller/main.tf]: Ah ok, thank you for making the changes here.

**Review by bryantbiggs — APPROVED** [review]

**Comment by nitrocode** [comment]
Thank you!!! Happy Friday!

**Pull request merged**